etc/service-bus.api.ts

```diff
@@ -0,0 +1,411 @@
+// @public
+interface AmqpError {
+  condition?: string;
+  description?: string;
+  info?: any;
+  value?: any[];
+}
+
+// @public
+interface ConnectionConfig {
+}
+
+// @public
+interface CorrelationFilter {
+  contentType?: string;
+  correlationId?: string;
+  label?: string;
+  messageId?: string;
+  replyTo?: string;
+  replyToSessionId?: string;
+  sessionId?: string;
+  to?: string;
+  userProperties?: any;
+}
+
+// @public
+interface DataTransformer {
+  decode: (body: any) => any;
+  encode: (body: any) => any;
+}
+
+// @public
+interface DeadLetterOptions {
+  deadLetterErrorDescription: string;
+  deadletterReason: string;
+}
+
+// @public
+class DefaultDataTransformer implements DataTransformer {
+  decode(body: any): any;
+  encode(body: any): any;
+}
+
+// @public
+export function delay<T>(t: number, value?: T): Promise<T>;
+
+// @public (undocumented)
+interface Delivery {
+  // (undocumented)
+  accept(): void;
+  // (undocumented)
+  readonly format: number;
+  // (undocumented)
+  readonly id: number;
+  // (undocumented)
+  readonly link: Sender | Receiver;
+  // (undocumented)
+  modified(params?: ReleaseParameters): void;
+  // (undocumented)
+  reject(error?: AmqpError): void;
+  // (undocumented)
+  release(params?: ReleaseParameters): void;
+  // (undocumented)
+  readonly remote_settled: boolean;
+  // (undocumented)
+  readonly remote_state?: DeliveryOutcome;
+  // (undocumented)
+  readonly sent: boolean;
+  // (undocumented)
+  readonly settled: boolean;
+  // (undocumented)
+  readonly state?: DeliveryOutcome;
+  // (undocumented)
+  readonly tag: Buffer | string;
+  // (undocumented)
+  update(settled: boolean, state?: any): void;
+}
+
+// @public
+interface Dictionary<T> {
+  // (undocumented)
+  [key: string]: T;
+}
+
+// @public (undocumented)
+interface MessageHandlerOptions {
+  autoComplete?: boolean;
+  maxAutoRenewDurationInSeconds?: number;
+  maxConcurrentCalls?: number;
+}
+
+// @public
+interface MessageHeader {
+  delivery_count?: number;
+  durable?: boolean;
+  first_acquirer?: boolean;
+  priority?: number;
+  ttl?: number;
+}
+
+// @public
+interface MessageProperties {
+  absolute_expiry_time?: number;
+  content_encoding?: string;
+  content_type?: string;
+  correlation_id?: string | number | Buffer;
+  creation_time?: number;
+  group_id?: string;
+  group_sequence?: number;
+  message_id?: string | number | Buffer;
+  reply_to?: string;
+  reply_to_group_id?: string;
+  subject?: string;
+  to?: string;
+  user_id?: string;
+}
+
+// @public
+class MessageSession extends LinkEntity {
+  // WARNING: The type "ClientEntityContext" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "MessageSessionOptions" needs to be exported by the package (e.g. added to index.ts)
+  constructor(context: ClientEntityContext, options?: MessageSessionOptions);
```

Comment:
ClientEntityContext need not be exported from the rollup. It is intended for internal use only.
The only way to get a MessageSession instance is to call `acceptSession()` method on the (Queue | Subscription)Client.

Reply:
MessageSessionOptions need not be exported since `acceptSession(options?: AcceptSessionOptions): Promise<MessageSession>` and `receiveMessgesFromSessions(onSessionMessage: OnSessionMessage, onError: OnError, options?: SessionHandlerOptions): Promise<void>;` on the QueueClient exposes different options that can be provided for creating MessageSession object.

Reply:
Does that mean that we need to remove the constructors that take those types? Or make them non-public?

Reply:
We can't remove those constructors. I wasnt aware of TS supporting private constructors. Just tried that now. Yes we should definitely make the constructor private.